### PR TITLE
Non-Authenticated Create New User

### PR DESCRIPTION
**Pull request by simonegiacco**
Solves #538

### CODE REVIEW

File: lib/endpoints/class-wp-rest-users-controller.php [hunk 1]
```diff
@@ -27,7 +27,7 @@ public function register_routes() {
 				'permission_callback' => array( $this, 'create_item_permissions_check' ),
 				'args'            => array_merge( $this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ), array(
 					'password'    => array(
-						'required' => true,
+						//'required' => true,
 					),
```
Comment (danielbachhuber) [inline lib/endpoints/class-wp-rest-users-controller.php:31]: We can remove this definition entirely.

File: lib/endpoints/class-wp-rest-users-controller.php [hunk 2]
```diff
@@ -285,21 +288,68 @@ public function create_item( $request ) {
...
+						  *
+						  * @since 4.4.0
+						  *
+						  * @param int $user_id ID of the newly created user.
+						  */
+						do_action( 'network_user_new_created_user', $user_id );
+					}
+				}
+			}else{
+				$user_id = register_new_user( $user->user_login, $user->user_email );
```
Comment (danielbachhuber) [inline lib/endpoints/class-wp-rest-users-controller.php:343]: We'll need to internalize the functionality for `register_new_user()`, as we won't be able to return HTML errors if registration fails.

**Comment by danielbachhuber** [comment]
@simonegiacco Can you add a few to several new tests covering this functionality, and the new success and error cases it provides?

File: lib/endpoints/class-wp-rest-users-controller.php (hunk 2 above)
Comment (simonegiacco) [inline lib/endpoints/class-wp-rest-users-controller.php:343]: ...and I guess strip_tags() is not an option,right? :)

Ok, I'll internalize the function, who can give me the error messages?

**Comment by danielbachhuber** [comment]
We're not going to be including this in the core plugin at this time, see https://github.com/WP-API/WP-API/pull/2356#issuecomment-195234273

**Pull request closed without merging**